### PR TITLE
Setting.py: Add end line number of setting

### PR DESCRIPTION
**Pull request by akshatkarani**
This adds end line number of a setting in coafile.

Related to https://github.com/coala/coala/issues/5044

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by akshatkarani** [comment]
@jayvdb  What do you think about finding end line number of a setting like this?

**Comment by akshatkarani** [comment]
@li-boxuan Can you please review this?

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot rebase

File: coalib/settings/Setting.py [hunk 1]
```diff
@@ -316,3 +317,12 @@ def line_number(self):
             raise TypeError("Instantiated with str 'origin' "
                             'which does not have line numbers. '
                             'Use SourcePosition for line numbers.')
+
+    @property
+    def end_line_number(self):
+        if isinstance(self._origin, SourcePosition):
+            return self.length + self._origin.line - 1
+        else:
+            raise TypeError("Instantiated with str 'origin' "
```
Comment (jayvdb) [inline coalib/settings/Setting.py]: Missing coverage of `326, 323->326`

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by akshatkarani** [comment]
What should be done next?

**Pull request merged**